### PR TITLE
script: subscription-manager support (part 2)

### PR DESCRIPTION
**Pull request by dachary**
Strip the .py suffix so that no pyc / pyo files are generated.

http://tracker.ceph.com/issues/14972 Fixes: #14972

Signed-off-by: Loic Dachary <loic@dachary.org>

### CODE REVIEW

**Comment by dachary** [comment]
followup of https://github.com/ceph/ceph/pull/7907

How can we make it so this is only installed on RHEL and not on CentOS where it serves no useful purpose ?

**Comment by branto1** [comment]
@dachary we would probably have to exclude centos with something like
`%if 0{?rhel} && ! 0{?centos}`
We should test it first, though.

**Comment by dachary** [comment]
in gitbuilders under wip-subman-facts

**Comment by tchaikov** [comment]
```
Checking for new repos for mirrors
error: parse error in expression
error: /tmp/install-deps.14358/ceph.spec:688: bad %if condition
Bad spec: /tmp/install-deps.14358/ceph.spec
```

failed, see http://gitbuilder.sepia.ceph.com/gitbuilder-ceph-rpm-centos7-amd64-basic/log.cgi?log=af63f27ab9bd5d2ac7d8c8e4518f7ed966aa5ee9

File: ceph.spec.in [hunk 1]
```diff
@@ -685,7 +685,7 @@ export RPM_OPT_FLAGS=`echo $RPM_OPT_FLAGS | sed -e 's/i386/i486/'`
                 --libexecdir=%{_libexecdir} \
 		--localstatedir=/var \
 		--sysconfdir=/etc \
-%if 0%{?rhel}
+%if 0{?rhel} && ! 0{?centos}
```
Comment (tchaikov) [inline ceph.spec.in]: maybe it should be
```
%if 0%{?rhel} && ! 0%{?centos}
```

**Comment by dachary** [comment]
@branto1 @tchaikov fixed and repushed, thanks !

**Comment by liewegas** [comment]
still has

Checking for unpackaged file(s): /usr/lib/rpm/check-files /srv/autobuild-ceph/gitbuilder.git/build/rpmbuild/BUILDROOT/ceph-10.0.5-2740.gf2db196.el7.x86_64
error: Installed (but unpackaged) file(s) found:
/etc/cron.hourly/subman

**Comment by tchaikov** [comment]
```
Checking for unpackaged file(s): /usr/lib/rpm/check-files /srv/autobuild-ceph/gitbuilder.git/build/rpmbuild/BUILDROOT/ceph-10.0.5-2740.gf2db196.el7.x86_64
error: Installed (but unpackaged) file(s) found:
/etc/cron.hourly/subman
```

we need to disable subman by default
```diff
--- a/configure.ac
+++ b/configure.ac
@@ -210,8 +210,7 @@ AM_CONDITIONAL(ENABLE_SERVER, test "$enable_server" = "yes")
 # subscription manager?
 AC_ARG_ENABLE([subman],
        [AS_HELP_STRING([--enable-subman], [enable subman])],
-       [],
-       [enable_subman=yes])
+       [enable_subman=$enableval])
 AM_CONDITIONAL([ENABLE_SUBMAN], test "$enable_subman" = "yes")
```

i just pushed a change with the above patch to wip-subman-facts to gitbuilder

**Comment by dachary** [comment]
@tchaikov thanks, I repushed an amended version

**Comment by tchaikov** [comment]
thank you @dachary , i understand the `AC_ARG_ENABLE` and `AC_ARG_WITH` macros better now.

**Comment by tchaikov** [comment]
lgtm. gitbuilders are happy.

**Pull request merged**